presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -130,8 +138,25 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Context> c
         {
             context.get().variables.addAll(node.getGroupingKeys());
 
-            for (AggregationNode.Aggregation aggregation : node.getAggregations().values()) {
-                aggregation.getArguments().forEach(expression -> subfieldExtractor.process(castToExpression(expression), context.get()));
+            for (VariableReferenceExpression variable : node.getAggregations().keySet()) {
+                AggregationNode.Aggregation aggregation = node.getAggregations().get(variable);
+                // Aggregate calls with pass-through functions should allow sub-field pruning
+                if (PASS_THROUGH_FUNCTIONS.contains(metadata.getFunctionManager().getFunctionMetadata(aggregation.getCall().getFunctionHandle()).getName())) {
+                    aggregation.getArguments().forEach(expression ->
+                    {
+                        Expression var = new SymbolReference(variable.getName());
+                        if (context.get().getPassThroughFields(var).isPresent()) {
```

Comment:
I'm not sure I understand this logic. Currently, there is only one pass-through aggregate function and it has only one argument. Hence, it is difficult for me to think though how would a multi-argument pass through function would work here. I suggest to simplify this change to work just for the `arbitrary` aggregate. I tried the following and the tests pass.

```
private static final QualifiedFunctionName ARBITRARY_AGGREGATE_FUNCTION = QualifiedFunctionName.of(DEFAULT_NAMESPACE, "arbitrary");

            for (Map.Entry<VariableReferenceExpression, AggregationNode.Aggregation> entry : node.getAggregations().entrySet()) {
                VariableReferenceExpression variable = entry.getKey();
                AggregationNode.Aggregation aggregation = entry.getValue();

                // Aggregate calls with pass-through functions should allow sub-field pruning
                QualifiedFunctionName aggregateName = metadata.getFunctionManager().getFunctionMetadata(aggregation.getCall().getFunctionHandle()).getName();
                if (ARBITRARY_AGGREGATE_FUNCTION.equals(aggregateName)) {
                    SymbolReference argument = (SymbolReference) castToExpression(aggregation.getArguments().get(0));
                    context.get().addAssignment(variable, new VariableReferenceExpression(argument.getName(), types.get(argument)));
                }
                else {
                    aggregation.getArguments().forEach(expression -> subfieldExtractor.process(castToExpression(expression), context.get()));
                }

                aggregation.getFilter().ifPresent(expression -> subfieldExtractor.process(castToExpression(expression), context.get()));

                aggregation.getOrderBy()
                        .map(OrderingScheme::getOrderByVariables)
                        .ifPresent(context.get().variables::addAll);

                aggregation.getMask().ifPresent(context.get().variables::add);
            }
```

Reply:
Thanks so much for the suggestion! We achieved the same with a lot less code. My understanding was not entirely correct and some of the code was duplicated effort of getting to what was already available in existing helper functions.